cpp/common/src/codingstandards/cpp/rules/identifierhidden/IdentifierHidden.qll

```diff
@@ -11,13 +11,36 @@ abstract class IdentifierHiddenSharedQuery extends Query { }
 
 Query getQuery() { result instanceof IdentifierHiddenSharedQuery }
 
-query predicate problems(UserVariable v2, string message, UserVariable v1, string varName) {
+/**
+ * There is a lambda that contains a declaration
+ * that hides something that is captured
+ * and the lambda exists in the function where this lamda is enclosed
```

Comment:
```suggestion
 * Holds if declaration `v2`, declared in a lambda, hides a declaration `v1` captured by the lambda.
```

Not that this is based on the current version. If the suggestions below are considered, we need to adjust this accordingly.